### PR TITLE
Dcs 2072 WPIP UI breadcrumbs

### PR DESCRIPTION
**Pull request by GurnankCheema**
Added breadcrumbs but switched to them back link via feature-flag for two pages in pre and prod

DEV
Home -> breadcrumb
People booked to arrive today  -> breadcrumb
Temporary Absences  -> breadcrumb

PRE & PROD
Home -> breadcrumb
People booked to arrive today  -> backlink
Temporary Absences  -> backlink

### CODE REVIEW

File: server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts [hunk 1]
```diff
@@ -13,12 +13,16 @@ let app: Express
 const temporaryAbsencesService = createMockTemporaryAbsencesService()
 const raiseAnalyticsEvent = jest.fn() as RaiseAnalyticsEvent
 
-beforeEach(() => {
+const createAppWithAllRoutes = () => {
```
Comment (GurnankCheema) [inline server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts:16]: to create the app in each test after setting different config.showBreadCrumb values

File: server/views/partials/layout.njk [hunk 2]
```diff
@@ -78,21 +78,27 @@
...
-                classes: "backLinkUrl",
-                attributes: {'data-qa': 'back-link'},
-                href: "#"
-            }) }}
-        {% endif %}
-
-        {% if breadCrumbs.enabled === true %}
-            {{ breadCrumb(breadCrumbs.pageTitle, breadCrumbs.data) }}
-        {% endif %}
+{% block beforeContent %}
```
Comment (GurnankCheema) [inline server/views/partials/layout.njk:81]: will eventually get rid of this confusing !== false logic across the app

File: server/views/pages/bookedtoday/choosePrisoner.njk [hunk 3]
```diff
@@ -1,6 +1,18 @@
 {% extends "../../partials/layout.njk" %}
 {% from  "../../partials/expectedArrivalCardMacro.njk" import expectedArrivalCard, transferCard  %}
 {% set pageTitle = "People booked to arrive today" if supportingMultitransactionsEnabled == true else "Select a person to add to the establishment roll" %}
+{% set hasDpsLink = false %}
+
+{% if showBreadCrumb %}
```
Comment (GurnankCheema) [inline server/views/pages/bookedtoday/choosePrisoner.njk:6]: showBreadCrumb is a temp config arg which we'll remove later on

File: server/routes/bookedtoday/choosePrisonerController.test.ts [hunk 4]
```diff
@@ -25,6 +25,36 @@ describe('GET /confirm-arrival/choose-prisoner', () => {
     expectedArrivalsService.getArrivalsForToday.mockResolvedValue(new Map())
   })
 
+  it('should only display back navigation odf the breadcrumb type', () => {
```
Comment (andrewrlee) [inline server/routes/bookedtoday/choosePrisonerController.test.ts:28]: minor typo

File: server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts [hunk 5]
```diff
@@ -36,7 +40,37 @@ describe('GET checkTemporaryAbsence', () => {
     return request(app).get('/prisoners/A1234AA/check-temporary-absence').expect(302).expect('Location', '/autherror')
   })
 
+  it('should only display back navigation of the back link type', () => {
+    config.showBreadCrumb = false
+    createAppWithAllRoutes()
```
Comment (andrewrlee) [inline server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts:45]: this is a bit gnarly - the var on app is a bit of a smell that suggests you have some unnecessary coupling - you need to call a method with a side effect or an apparently unrelated variable (`app`) will be invalid. 
 
Would suggest getting rid of `app` variable and returning it from  `createAppWithAllRoutes()` passing and moving breadcrumb setup into there: 

so you can just do something like:

`return request(createApp({breadcrumb: true}))`

where create app looks like: 
```
const createApp = ({ breadcrumb}: {breadcrumb: boolean}) => {
   config.showBreadCrumb = breadcrumb
  return appWithAllRoutes({
    services: { temporaryAbsencesService, raiseAnalyticsEvent },
    roles: [Role.PRISON_RECEPTION],
  })
}

```

File: server/views/pages/bookedtoday/summaryMoveOnly.njk [hunk 6]
```diff
@@ -4,6 +4,8 @@
 
 {% set pageTitle = "Summary for person booked to arrive" %}
 {% set hasBackLink = false %}
+{% set hasDpsLink = false %}
```
Comment (andrewrlee) [inline server/views/pages/bookedtoday/summaryMoveOnly.njk:7]: is this the wrong way round - it seems like most pages don't have the dps back link?
Would it be better to default to not having the link and only set the variable when you do have it?

Comment (GurnankCheema) [reply server/views/pages/bookedtoday/summaryMoveOnly.njk:7]: yes that would make more sense

File: server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts (hunk 5 above)
Comment (GurnankCheema) [reply server/routes/temporaryabsences/checkTemporaryAbsenceController.test.ts:45]: I did think of doing that at first but then felt it was more readable having a separate variable.

**Pull request merged**